### ISSUE TITLE
Derailed for no reason

### ISSUE THREAD

**Issue by navaroneee** [EternalBlueFlame/Traincraft-5#291]
Good evening, first of all I would like to congratulate you for the excellent work done so far.
Now, let's go to the problems I've been observing, when I pull a wagon from the rails, they duplicate, we play on a server, I do not know if it interferes with something,
Second error: i train seems to derail for no apparent reason, sometimes I'm at speeds less than 20KM / H and it simply sits sideways, sometimes even backwards.
Thank you very much in advance

Version: Traincraft-4.3.2_009
Server: Thermos-1.7.10-1614-58
Forge: 10.13.4.1614

Ps: sorry for my bad english :P

**Comment by EternalBlueFlame** [comment]
The duplication issue has already been fixed and is awaiting 010 release.

As for the derail issue, is this on any specific rail?
The calculations are different for each one, so either it's a block detection issue, or it's something with how the train uses that one rail.

**Comment by navaroneee** [comment]
Great news, duplication is a little bit OP hahahh

I have notice that happens in 5x5 curve rail, but in 3x3 happens too.
I think 9x9 never do this. I tested it with different speeds, it seems to be random, sometimes it derail with 20km/h other times it only derail at 60km/h, on the same track,

**Comment by EternalBlueFlame** [comment]
Interesting... Is this with trains other than the ic4? I know that causes it due to being longer than TC's current code supports (that will also be fixed in 010).

Honestly longer trains are gonna be more prone to the issue due to shoddy pathfinding.
But if it happens with most trains, then there's a bigger problem.

**Comment by navaroneee** [comment]
I tested with "Santa Fe" and "Union Pacific".
It looks like it hits a block, the locomotive is on its side and the wagons come loose, but there is no message of derailment.
Another issue I noticed is that when the chunks clear the wagons "get into each other" (img below)
![2017-06-15_23 47 03](https://user-images.githubusercontent.com/24437327/27240979-31ef6d06-52ad-11e7-8339-f4a56c07b38e.png)

**Comment by EternalBlueFlame** [comment]
Yeah, the chunk reset bug has been around since the 1.7.10 port started, the chunkloading patch fixed it to a degree, but not fully. I'm still not entirely sure what's going wrong aside from maybe missing NBT data, or bad position reset code, but that's just a guess, and it's difficult to track down.

As for the "it hits a block" it may be something to do with the ray trace of blocks... it might be colliding with an invisible block of some form, like Railcraft's residual heat block, or TiM's dynamic lighting block... although that _shouldn't_ be possible.
It could also be going from one turn directly into another with no straight inbetween, which the manual warns against because of the shoddy pathfinding.
It could also be something related to desync... Can you replicate the issue in singleplayer? With or without railcraft I can't seem to replicate it, but I can only test singleplayer, which natrually doesn't desync.

![java_2017-06-16_19-42-59](https://user-images.githubusercontent.com/9969122/27248665-8a9a19aa-52cc-11e7-8ea5-63f37c5e99ed.png)

I also changed the label since the duplication issue was fixed, hopefully help prevent some confusion for others, hope you don't mind.

**Comment by navaroneee** [comment]
"I also changed the label since the duplication issue was fixed, hopefully help prevent some confusion for others, hope you don't mind." No problem :D 

"Can you replicate the issue in singleplayer?" 
I'm going on a trip now, and I'm going back Sunday afternoon, I'm going to test the singleplayer and post the results, but I have a testing world on the single and I believe it really did not happen

**Comment by EternalBlueFlame** [comment]
Alright, take your time, there's still plenty of time before the 010 release to get it (maybe) fixed, knowing our artists.

**Comment by EternalBlueFlame** [comment]
Another thought (ItsNimnom reminded me), double check to be sure chunkloading is enabled, because the trains use that a lot for pathfinding and linking, especially on servers.
There's a forge config so you can limit it's abilities on a per-mod basis, which may help if you're running the server on a potato computer.

**Comment by navaroneee** [comment]
I did some tests online and offline, it seems that the problem happens more with some entity this in the train, either player or mobs, I did all the way out of the train and no problem happened, when I was in the locomotive happened again, I also tested putting A cow in a wagon, and it happened again. (Offiline happened only once, with me in the locomotive and with the cow in the wagon)
